cli/ballerina-launcher/src/main/java/org/ballerinalang/launcher/LauncherUtils.java

```diff
@@ -94,18 +94,28 @@ public static void runProgram(Path sourceRootPath, Path sourcePath, String funct
         System.setProperty(ProjectDirConstants.BALLERINA_SOURCE_ROOT, fullPath.getParent().toString());
         loadConfigurations(fullPath.getParent(), runtimeParams, configFilePath, observeFlag);
 
+        boolean isProject = RepoUtils.hasProjectRepo(sourceRootPath);
         if (srcPathStr.endsWith(BLANG_EXEC_FILE_SUFFIX)) {
             programFile = BLangProgramLoader.read(sourcePath);
-        } else if (Files.isRegularFile(fullPath) &&
-                srcPathStr.endsWith(BLANG_SRC_FILE_SUFFIX) &&
-                !RepoUtils.hasProjectRepo(sourceRootPath)) {
+        } else if (Files.isRegularFile(fullPath) && srcPathStr.endsWith(BLANG_SRC_FILE_SUFFIX) && !isProject) {
             programFile = compile(fullPath.getParent(), fullPath.getFileName(), offline);
         } else if (Files.isDirectory(sourceRootPath)) {
-            if (Files.isDirectory(fullPath) && !RepoUtils.hasProjectRepo(sourceRootPath)) {
+            // If we are trying to run a package from outside a project
+            if (Files.isDirectory(fullPath) && !isProject) {
                 throw LauncherUtils.createLauncherException(
                         "error: did you mean to run the Ballerina package as a project? If so run 'ballerina init' to" +
                                 " make it a project with a .ballerina directory");
```

Comment:
Run either from within the project or use sourceroot.

Reply:
Fixed with 449b8a6